hooks/useTreasuryInfo/getTokenOwnerRecordsForWallet.tsx

```diff
@@ -0,0 +1,158 @@
+import { MintInfo } from '@blockworks-foundation/mango-v4'
+import { getCertifiedRealmInfos, RealmInfo } from '@models/registry/api'
+import { AssetType, TokenOwnerRecordAsset } from '@models/treasury/Asset'
+import {
+  Governance,
+  GovernanceAccountParser,
+  GOVERNANCE_PROGRAM_SEED,
+  ProgramAccount,
+  Realm,
+  TokenOwnerRecord,
+} from '@solana/spl-governance'
+import { AccountInfo, PublicKey } from '@solana/web3.js'
+import { ConnectionContext } from '@utils/connection'
+import { parseMintAccountData, TokenProgramAccount } from '@utils/tokens'
+import axios from 'axios'
+
+const getAccountInfoFromRaw = (raw: any) => {
+  const accountInfo: AccountInfo<Buffer> = {
+    executable: raw.executable,
+    lamports: raw.lamports,
+    owner: new PublicKey(raw.owner),
+    rentEpoch: raw.rentEpoch,
+    data: Buffer.from(raw.data[0], 'base64'),
+  }
+  return accountInfo
+}
+
+const getTokenOwnerRecordsForWallet = async (
+  connection: ConnectionContext,
+  governanceAccount: ProgramAccount<Governance>,
+  walletAddress?: PublicKey | null
+): Promise<TokenOwnerRecordAsset[]> => {
+  if (!walletAddress) return []
+
+  const realmInfos = getCertifiedRealmInfos(connection)
+
+  const cacheMap: Record<
+    number,
+    { tokenOwnerRecord: PublicKey; realmInfo: RealmInfo }
+  > = {}
+
+  const jsonRpcRequestData = realmInfos
+    .map((realm, i) => {
+      if (!realm.communityMint) return null
+
+      const [tokenOwnerRecordAddress] = PublicKey.findProgramAddressSync(
+        [
+          Buffer.from(GOVERNANCE_PROGRAM_SEED),
+          realm.realmId.toBuffer(),
+          realm.communityMint.toBuffer(),
+          walletAddress.toBuffer(),
+        ],
+        realm.programId
+      )
+
+      cacheMap[i] = {
+        tokenOwnerRecord: tokenOwnerRecordAddress,
+        realmInfo: realm,
+      }
+
+      return {
+        jsonrpc: '2.0',
+        id: i,
+        method: 'getMultipleAccounts',
+        params: [
+          [
+            realm.realmId.toBase58(),
+            tokenOwnerRecordAddress.toBase58(),
+            realm.communityMint.toBase58(),
+          ],
+          {
+            commitment: connection.current.commitment,
+            encoding: 'base64',
+          },
+        ],
+      }
+    })
+    .filter((x) => x !== null)
+
+  try {
+    const rawAccountData = await axios.request({
+      url: connection.endpoint,
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      data: JSON.stringify(jsonRpcRequestData),
+    })
```

Comment:
@abrzezinski94 I've implemented batching of rpc request here, where its a single RPC request to fetch an array of [realmAccount, tokenOwnerRecordAccount, communityMint] for each of the possible daos.